sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -874,20 +866,24 @@ fn init_modules<Mode: ExecutionMode>(
     });
 
     for module_id in modules_to_init {
-        let return_values = execute_move_call::<Mode>(
+        match execute_move_call::<Mode>(
             context,
             argument_updates,
             &module_id,
             INIT_FN_NAME,
             vec![],
             vec![],
             /* is_init */ true,
-        )?;
-
-        assert_invariant!(
-            return_values.is_empty(),
-            "init should not have return values"
-        )
+        ) {
+            Ok(return_values) => assert_invariant!(
+                return_values.is_empty(),
+                "init should not have return values"
+            ),
+            Err(err) => {
+                context.linkage_view.unpublish_package(modules);
```

Comment:
Woah...

Reply:
Is this new? Or something the session automatically did?

Reply:
yeah that is the part of session we were using and I just shoveled it there for now.
Not clear it will stay there.
But basically module init needed that code. I plan to review the ordeal of module initializers so we'll see...
but yeah that is the new non-sense.
Maybe we should fined a better way to do that. Thoughts? Feelings?

Reply:
in fact come to think of it this is a question for @amnn on how he may want to get this done.
I added the list of modules here just because it was simple and easy. But we have the list of published packages in context, maybe we should reconcile?

Reply:
Moved the `DataStore` implementation close to the `ExecutionContext` and the new packages tracked there